bitbucket-scm-filter-aged-refs/src/main/java/org/jenkinsci/plugins/scm_filter/BitbucketAgedRefsTrait.java

```diff
@@ -19,26 +19,27 @@
 import org.jenkinsci.Symbol;
 import org.kohsuke.stapler.DataBoundConstructor;
 
-/**
- * @author witokondoria
- */
 public class BitbucketAgedRefsTrait extends AgedRefsTrait {
 
     /**
      * Constructor for stapler.
      *
-     * @param retentionDays retention period in days
+     * @param branchRetentionDays retention period in days for branches
+     * @param prRetentionDays     retention period in days for pull requests
+     * @param tagRetentionDays    retention period in days for tags
+     * @param branchExcludeFilter space-separated list of branch name patterns to
+     *                            ignore. For example: release main hotfix-*
      */
     @DataBoundConstructor
-    public BitbucketAgedRefsTrait(String retentionDays) {
-        super(retentionDays);
+    public BitbucketAgedRefsTrait(
```

Comment:
Please don't add new oprtions to the `@DataBoundConstructor`, use `@DataBoundSetter`s instead.

Reply:
I'm not a Jenkins plugin expert, I am assuming you don't want to break the deserialization of existing `config.xml` files that don't have a value for the new parameters? The problem is there is no more `retentionDays` variable. So what should we do?

What if I changed it to...

```
    public BitbucketAgedRefsTrait() {
        super();
    }

    @DataBoundSetter
    public void setBranchRetentionDays(String branchRetentionDays) {
        super.setBranchRetentionDays(branchRetentionDays);
    }

    @DataBoundSetter
    public void setPrRetentionDays(String prRetentionDays) {
        super.setPrRetentionDays(prRetentionDays);
    }

    @DataBoundSetter
    public void setTagRetentionDays(String tagRetentionDays) {
        super.setTagRetentionDays(tagRetentionDays);
    }

    @DataBoundSetter
    public void setBranchExcludeFilter(String branchExcludeFilter) {
        super.setBranchExcludeFilter(branchExcludeFilter);
    }
```

Reply:
If you removed a variable, it needs to stay in the class to not break the serialization format, in the base case with migration code to new variables... (Or keep the coarse-grained setting and hide the individual settings behind an "advanced" pop-out?)

Reply:
See https://www.jenkins.io/doc/developer/persistence/backward-compatibility/

Reply:
As far as I see you only have to add something like:
```
protected transient Long retentionDays;

protected Object readResolve() {
  if (retentionDays!= null) {
    super.setBranchRetentionDays(retentionDays); 
    super.setPrRetentionDays(retentionDays);
    super.setTagRetentionDays(retentionDays);
    super.setBranchExcludeFilter(retentionDays);
  }
  return this;
}
```

Reply:
If I follow ☝️  do I revert my changes to `@DataBoundConstructor`?

Reply:
Ok well I applied the changes based on feedback and the doc linked.